### PR TITLE
feat: adjust reserve tokens, supply tokens, borrow tokens from logics

### PR DESCRIPTION
**Pull request by LuPoYi**
- [x] adjust reserve tokens, supply tokens, borrow tokens from logics
- [x] unit test (local)
- [x] yarn changeset

### CODE REVIEW

File: packages/lending/src/protocols/aave-v2/lending-protocol.ts [hunk 1]
```diff
@@ -67,21 +59,23 @@ export class LendingProtocol extends Protocol {
...
 
-    const reserveMap: Record<string, Reserve> = {};
+    for (const reserveToken of reserveTokens) {
+      const { asset, aToken } = reserveToken;
 
-    for (const { asset, aToken, stableDebtToken, variableDebtToken } of reserves) {
       if (asset.isWrapped) {
-        reserveMap[asset.unwrapped.address] = { aToken, asset };
-        reserves.push({ asset: asset.unwrapped, aToken, stableDebtToken, variableDebtToken });
+        reserveMap[asset.unwrapped.address] = { ...reserveToken, asset: asset.unwrapped };
```
Comment (Jeff-CCH) [inline packages/lending/src/protocols/aave-v2/lending-protocol.ts]:
```suggestion
        reserveMap[asset.unwrapped.address] = { ...reserveToken, asset: asset};
```
should be wrapped?

File: packages/lending/src/protocols/aave-v3/configs.ts [hunk 2]
```diff
@@ -1,14 +1,11 @@
...
 import * as logics from '@protocolink/logics';
 
 export const ID = 'aave-v3';
 export const DISPLAY_NAME = 'Aave V3';
 
-export interface Reserve {
-  asset: common.Token;
-  aToken: common.Token;
-}
+export type ReserveTokens = logics.aavev2.ReserveTokens;
```
Comment (Jeff-CCH) [inline packages/lending/src/protocols/aave-v3/configs.ts]:
```suggestion
export type ReserveTokens = logics.aavev3.ReserveTokens;
```
should be aave v3?

File: packages/lending/src/protocols/aave-v3/lending-protocol.ts [hunk 3]
```diff
@@ -63,23 +55,25 @@ export class LendingProtocol extends Protocol {
...
 
-    for (const { asset, aToken, stableDebtToken, variableDebtToken } of reserves) {
+    for (const reserveToken of reserveTokens) {
+      const { asset, aToken } = reserveToken;
       if (asset.isNative) hasNativeToken = true;
 
       if (asset.isWrapped) {
-        reserveMap[asset.unwrapped.address] = { aToken, asset };
-        reserves.push({ asset: asset.unwrapped, aToken, stableDebtToken, variableDebtToken });
+        reserveMap[asset.unwrapped.address] = { ...reserveToken, asset: asset.unwrapped };
```
Comment (Jeff-CCH) [inline packages/lending/src/protocols/aave-v3/lending-protocol.ts]:
```suggestion
        reserveMap[asset.unwrapped.address] = { ...reserveToken, asset: asset};
```

File: packages/lending/src/protocols/morphoblue/adapter.test.ts [hunk 4]
```diff
@@ -273,7 +273,7 @@ describe('Test Adapter for Morpho Blue', async function () {
   // leverage only have one scenario - leverage by collateral token
   context('Test leverageByCollateral', function () {
     const account = '0x9edcb464C0AfdD01a5Ffbd09309b437C7dadeAB3';
-    const blockTag = 20180000;
+    const blockTag = 20380000;
```
Comment (Jeff-CCH) [inline packages/lending/src/protocols/morphoblue/adapter.test.ts:276]: is there any reason to change this blockTag?

File: packages/lending/src/protocols/radiant-v2/lending-protocol.ts [hunk 5]
```diff
@@ -62,20 +54,23 @@ export class LendingProtocol extends Protocol {
...
+    const reserveMap: ReserveMap = {};
+
+    for (const reserveToken of reserveTokens) {
+      const { asset, rToken } = reserveToken;
 
-    for (const { asset, rToken, stableDebtToken, variableDebtToken } of reserves) {
       if (asset.isWrapped) {
-        reserveMap[asset.unwrapped.address] = { rToken, asset };
-        reserves.push({ asset: asset.unwrapped, rToken, stableDebtToken, variableDebtToken });
+        reserveMap[asset.unwrapped.address] = { ...reserveToken, asset: asset.unwrapped };
```
Comment (Jeff-CCH) [inline packages/lending/src/protocols/radiant-v2/lending-protocol.ts]:
```suggestion
        reserveMap[asset.unwrapped.address] = { ...reserveToken, asset: asset};
```

File: packages/lending/src/protocols/radiant-v2/lending-protocol.test.ts [hunk 6]
```diff
@@ -198,149 +197,252 @@ describe('Test Radiant V2 LendingProtocol', function () {
...
-              price: '62129.33',
-              balance: '114.493054935484124489',
-              apy: '0.0018842556485471865',
+              token: {
+                chainId: 56,
+                address: '0x7130d2A12B9BCbFAe4f2634d864A1Ee1Ce3Ead9c',
+                decimals: 18,
+                symbol: 'BTCB',
+                name: 'BTCB Token',
+                logoUri: 'https://cdn.furucombo.app/assets/img/token/WBTC.svg',
```
Comment (Jeff-CCH) [inline packages/lending/src/protocols/radiant-v2/lending-protocol.test.ts:218]: Do you think it's better to get the token information from other file/package like the old one OR it's better to have a fixed value here?

File: packages/lending/src/protocols/spark/configs.ts [hunk 7]
```diff
@@ -1,14 +1,11 @@
...
 import * as logics from '@protocolink/logics';
 
 export const ID = 'spark';
 export const DISPLAY_NAME = 'Spark';
 
-export interface Reserve {
-  asset: common.Token;
-  aToken: common.Token;
-}
+export type ReserveTokens = logics.aavev2.ReserveTokens;
```
Comment (Jeff-CCH) [inline packages/lending/src/protocols/spark/configs.ts]:
```suggestion
export type ReserveTokens = logics.spark.ReserveTokens;
```

File: packages/lending/src/protocols/spark/lending-protocol.ts [hunk 8]
```diff
@@ -63,23 +55,25 @@ export class LendingProtocol extends Protocol {
...
 
-    for (const { asset, aToken, stableDebtToken, variableDebtToken } of reserves) {
+    for (const reserveToken of reserveTokens) {
+      const { asset, aToken } = reserveToken;
       if (asset.isNative) hasNativeToken = true;
 
       if (asset.isWrapped) {
-        reserveMap[asset.unwrapped.address] = { aToken, asset };
-        reserves.push({ asset: asset.unwrapped, aToken, stableDebtToken, variableDebtToken });
+        reserveMap[asset.unwrapped.address] = { ...reserveToken, asset: asset.unwrapped };
```
Comment (Jeff-CCH) [inline packages/lending/src/protocols/spark/lending-protocol.ts]:
```suggestion
        reserveMap[asset.unwrapped.address] = { ...reserveToken, asset: asset};
```

File: packages/lending/src/protocols/morphoblue/adapter.test.ts (hunk 4 above)
Comment (LuPoYi) [reply packages/lending/src/protocols/morphoblue/adapter.test.ts:276]: not quite sure why the old blockTag could make it fail while `getPortfolio`

File: packages/lending/src/protocols/radiant-v2/lending-protocol.test.ts (hunk 6 above)
Comment (LuPoYi) [reply packages/lending/src/protocols/radiant-v2/lending-protocol.test.ts:218]: Directly using fixed values might be more efficient, tests might still need to be updated manually whenever new tokens are added

File: packages/lending/src/protocols/aave-v2/lending-protocol.ts (hunk 1 above)
Comment (LuPoYi) [reply packages/lending/src/protocols/aave-v2/lending-protocol.ts]: fixed, we keep the same logic as before

File: packages/lending/src/protocols/aave-v3/configs.ts (hunk 2 above)
Comment (LuPoYi) [reply packages/lending/src/protocols/aave-v3/configs.ts]: Apologies, a typo.

File: packages/lending/src/protocols/aave-v3/lending-protocol.ts (hunk 3 above)
Comment (LuPoYi) [reply packages/lending/src/protocols/aave-v3/lending-protocol.ts]: fixed

File: packages/lending/src/protocols/radiant-v2/lending-protocol.ts (hunk 5 above)
Comment (LuPoYi) [reply packages/lending/src/protocols/radiant-v2/lending-protocol.ts]: fixed

File: packages/lending/src/protocols/spark/configs.ts (hunk 7 above)
Comment (LuPoYi) [reply packages/lending/src/protocols/spark/configs.ts]: fixed

File: packages/lending/src/protocols/spark/lending-protocol.ts (hunk 8 above)
Comment (LuPoYi) [reply packages/lending/src/protocols/spark/lending-protocol.ts]: fixed

**Review by Jeff-CCH — APPROVED** [review]
LGTM

**Pull request merged**